### DISCUSSION TITLE
Pybricks, now with blocks!

### DISCUSSION THREAD

**Discussion by laurensvalk** [pybricks/support#1313] [Announcements]
## Endless creativity and fun with smart LEGO® bricks using Pybricks

![pybricks-blocks-2](https://github.com/pybricks/support/assets/12326241/55b4b2c5-923d-4812-8dad-8fb79efd18c8)

- For the first time ever, fans of all LEGO themes can bring their smart bricks together in a single app for endless possibilities and creativity.
- Make your LEGO City trains, LEGO BOOST creatures, LEGO Technic machines, or SPIKE and MINDSTORMS robots come alive. All with the same code blocks!
- Programs are saved directly onto the LEGO hubs. Just press the button and go. No permanent connections required.
- Hubs communicate wirelessly so you can build at every scale.
- Accurate vehicle control using built-in gyro for robot missions.
- Smooth transition from blocks to Python with live preview.
- Low floors and high ceilings: Easy to get started, but endless possibilities by mixing blocks with Python.
- Exclusively available to Pybricks supporters on Patreon.

**November 28th, 2023 - Pybricks Headquarters**: This week, the Pybricks team presents the first beta release of block coding for all modern LEGO® hubs. For the first time, fans of all LEGO themes can bring their smart bricks together in a single app for endless possibilities and creativity.

Whether you want to make smart train layouts, autonomous Technic machines, interactive BOOST creatures, or super-precise SPIKE and MINDSTORMS robots, you can do it with Pybricks.

Pybricks is beginner-friendly and easy to use. There’s no need to install complicated apps or libraries either. Just go to https://beta.pybricks.com/, update the firmware, and start coding.

And now for the first time, no prior Python coding experience is required. You can code with familiar but powerful blocks, and gradually switch to Python when you’re ready. The live preview makes it easy to see how your blocks translate to Python code.

Meanwhile, more seasoned builders and robotics teams will enjoy advanced features such as color sensor calibration or builtin gyro control for drive bases.

The new block coding experience is exclusively available to our supporters on [Patreon](https://www.patreon.com/pybricks). You can sign up for a monthly subscription or make a one-time pledge in our [shop](https://www.patreon.com/pybricks/shop) for lifetime access. If you already sponsored Pybricks on GitHub prior to today, you get instant access too of course! See the note at the bottom of this post.

Python coding remains entirely free and open source, and continues to be supported by a community of developers and LEGO enthusiasts around the world. Improvements are made almost every day, with the lead developers actively engaging with the community for ideas, bug fixes, and brand new features.

So grab your LEGO sets and [start coding](https://beta.pybricks.com/)!

See more info and videos in the [press release](https://pybricks.com/about/new-pybricks-blocks/).

![codepreview](https://github.com/pybricks/support/assets/12326241/67499ed4-6344-489e-a0c7-4ff2a9016ff3)
![multitask](https://github.com/pybricks/support/assets/12326241/3f6816ee-04ec-424e-a4fb-4ab8f2da179d)
![devices](https://github.com/pybricks/support/assets/12326241/07489551-07bb-4605-be02-3929109d26fe)
![pybricks-blocks-docs](https://github.com/pybricks/support/assets/12326241/214be00c-8c66-415b-a930-365ce924fec7)
![builder](https://github.com/pybricks/support/assets/12326241/555fa1af-60fe-4120-bd81-21f956b620ac)

**Direct access for existing and prior GitHub sponsors**

We'd like to thank everyone who sponsored the project so far! All our monthly GitHub sponsors prior to this date (no matter the amount) can sign up for free over at Patreon as shown below. If you were a one-off sponsor, we'll set you up for a matching duration as well. Please email us at team@pybricks.com so we know which email address to give access to. Thanks!

![image](https://github.com/pybricks/support/assets/12326241/32d324b3-0feb-4554-80c9-672af6db1c9f)

**Comment by albjerre** [comment]
How does the licensing work with the Patreon monthly subscription? Will it like with the perpetual license also only be one computer, you can be logged in to at once?

**Reply by laurensvalk** [reply]
In both cases, you can use them on multiple devices for personal use. For example on your computer and phone.

**Reply by albjerre** [reply]
Ok. But do you in both cases have to deactivate a license on a computer before using it on another computer as it is described at the top here?
https://pybricks.onfastspring.com/

**Reply by laurensvalk** [reply]
It is not necessary to deactivate on your PC if you also want to use it on your phone. Both are fine for personal use.

Really the main thing is to not redistribute them to others. Because we want to keep things flexible - we appreciate your support! :)

**Reply by albjerre** [reply]
Ok. Thanks. I think I will buy a perpetual license :)

**Comment by seralex807017** [comment]
how to save remote control program on hub

**Reply by laurensvalk** [reply]
Any program you run is automatically saved on the hub. Feel free to start a new discussion for further questions.

**Comment by laurettamorris** [comment]
Hi, Not sure if I am posting to the correct location.  I am a school robotics coach and this is our first year suing Pybricks so we are using the blocks. 

I am creating my own function and trying to pass a parameter to it.  I am not finding a block being created anywhere that will allow me to access the passed parameter.  

Can  you please point me in the right direction for this to find a solutions?

Thanks! Loving pybricks so far!

Lauretta

**Reply by DylanBPY** [reply]
I've never personally used the blocks, but I'm pretty sure you can just drag the parameter.

**Comment by quakec** [comment]
Huge thank you for making this project happen, and for providing a perpetual license at a reasonable price! It works seamlessly with the newest Spike devices and is far better than the "education" orientated lego original. The code blocks are useful to get started out in Python as you can see the code being generated side-by-side. Beautiful idea!

**Comment by Eggmanplant** [comment]
What is the license situation? The web editor still says it's open source, but the repo is severely outdated.
I will not use the blocks for now, as I hate web apps and proprietary software.

**Reply by laurensvalk** [reply]
Except for the blocks, everything is up to date and clearly licensed.

> I will not use the blocks for now, as I hate web apps and proprietary software.

Thanks for your input.

The block editor is a completely optional add-on that funds our open source work, so that users like you can enjoy using it for free.

**Comment by seralex8070-spec** [comment]
Hello, please tell me how I can make a program in Pybricks blocks for a car with three motors on the wheels. The standard DriveBase block does not allow adding another motor (for example, for a 6x6 truck). And if I go to the window on the right where the text version of my program is, then when adding a rotor manually, it tells me that this is for viewing only and cannot be edited.

**Reply by albjerre** [reply]
> Hello, please tell me how I can make a program in Pybricks blocks for a car with three motors on the wheels. The standard DriveBase block does not allow adding another motor (for example, for a 6x6 truck). And if I go to the window on the right where the text version of my program is, then when adding a rotor manually, it tells me that this is for viewing only and cannot be edited.

The DriveBase is only designed for two motors. You have to manually recreate the DriveBase functionality if you want three motors. How are you planning to use the three motors?

**Reply by laurensvalk** [reply]
I think this is about the `Car` block. I've opened a feature request here: https://github.com/pybricks/support/issues/2686

**Reply by seralex8070-spec** [reply]
This is an all-terrain vehicle with three axles and each axle has its own L motor, and another motor for steering and all this beauty must be controlled with an Xbo
<img width="4032" height="2268" alt="PXL_20260428_190255912 MP" src="https://github.com/user-attachments/assets/86e0ea63-756a-487b-a1c0-fac19fc110a9" />
x controller.

**Reply by seralex8070-spec** [reply]
Can someone tell me how to do this with additional blocks?

**Reply by dlech** [reply]
I wonder if you could make an extra task that continuously reads the speed (or position?) of one motor that is part of the `Car` class and tells the other motor to mirror it.

**Reply by laurensvalk** [reply]
Good idea @dlech. And if you use "drive at power" you can just add the extra motor yourself:

<img width="1060" height="922" alt="image" src="https://github.com/user-attachments/assets/f4700e05-78cd-4492-8f60-ea89a11118f2" />
